### DISCUSSION TITLE
generate Ed25519 keys

### DISCUSSION THREAD

**Discussion by dev-area** [polhenarejos/pico-hsm#30] [Q&A] [answered]
hi
couldnt find a way to create ed25519 key using sc-tool 
I tried:

sc-tool -l --pin 648219 --keygen --key-type EC:edwards25519 --id 12 --label "edtest"

getting unknown key type
any idea?

thanks

**Comment by polhenarejos** [comment, marked as answer]
You cannot use the module provided by OpenSC/SmartCard (namely sc-tool or pkcs11-tool) because this driver does not support Edward curves. Originally this driver is intented to work only with smartcard hsm from cardcontact, which does not support Edwards.

If you want to generate Ed25519/Ed448 curves you must use supplied tool `pico-hsm-tool`.

**Reply by dev-area** [reply]
thanks

**Reply by polhenarejos** [reply]
Clone the repo and type `pip install .` (note the dot `.`) inside the `pypicohsm` folder.

**Reply by dev-area** [reply]
thanks , 
I found it 2 minutes after posting my question so I removed it but you are fast :)

**Comment by nimbius** [comment]
fascinating...so just to confirm we _can_ generate ed25519?  

are you accepting pulls for https://www.picokeys.com/pico-hsm/  ? we should update the featurelist to include this if so :)

**Reply by polhenarejos** [reply]
Yes, it can generate ed25519 but it is in beta stage. I sent some PR to MbedTLS but they are in review since two years, so it is progressing slowly. As soon they get approved, I'll mark Pico HSM with Ed25519 as stable.

If you have a PR for Pico HSM, please send it and I'll review it.

**Comment by polhenarejos** [comment]
`development-eddsa` has support for Ed25519 key generation via `pico-hsm-tool`.
